src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignedCms.cs

```diff
@@ -557,7 +557,7 @@ internal void UpdateCertificatesFromAddition(X509Certificate2Collection newCerts
 
             if (existingLength > 0 || newCerts.Count > 1)
             {
-                var certs = new HashSet<X509Certificate2>(Certificates.OfType<X509Certificate2>());
+                var certs = new HashSet<X509Certificate2>(Certificates);
```

Comment:
The `System.Security.Cryptography.Pkcs` package is still built against .NET Framework (which won't have the new interface implementations).  You won't be able to remove the `OfType` from projects that build against anything other than `NetCoreAppCurrent`

https://github.com/dotnet/runtime/blob/2df392394d3a8d88b451486ea1dd983b70ed9e0d/src/libraries/System.Security.Cryptography.Pkcs/src/System.Security.Cryptography.Pkcs.csproj#L7

It's probably fine to leave the `OfType` and other things in the `Pkcs` package.